client/src/app/components/common/new-file-modal.component.ts

```diff
@@ -113,15 +118,17 @@ export class NewFileModalComponent {
         Observable.of(1).delay(50).filter(_ => this.error === undefined).subscribe(_ => this.isCreatingFile = true);
 
         const fileType = this.fileTypeRadio.getSelectedValue();
-        let path       = new FileName(form.controls["filename"].value);
+        let fileName   = new FileName(form.controls["filename"].value);
 
         if (fileType === "workflow" || fileType === "command_line_tool") {
-            path = path.ensureExtension("json");
+            fileName = fileName.ensureExtension("json");
```

Comment:
Just noticed this `ensureExtension("json")` detail. As far as draft-2 is concerned: yes, newly created files should be JSON because that's currently all SB supports and we want to be SB compliant.

However, v1.0 should default to YAML because it's easier to work with textually, especially with the v1.0 syntactic sugars.